projects/train/train/data/base.py

```diff
@@ -68,7 +69,7 @@ def __init__(
         trigger_pad: float = 0,
         fftlength: Optional[float] = None,
         highpass: Optional[float] = None,
-        snr_sampler: Optional[Callable[[int], torch.Tensor]] = None,
+        snr_sampler: Optional[TransformedDist] = None,
```

Comment:
Main reason I left the type hint so general was to support our old `SnrSampler` which I don't think jsonargparse would treat as compatible here

Reply:
True, I can include the other types as well